src/Stratis.Bitcoin.IntegrationTests/Wallet/WalletAddressGenerationAndFundsVisibility_Steps.cs

```diff
@@ -0,0 +1,130 @@
+using Stratis.Bitcoin.IntegrationTests.Common;
+using Stratis.Bitcoin.IntegrationTests.Common.Builders;
+using Stratis.Bitcoin.IntegrationTests.Common.EnvironmentMockUpHelpers;
+using Stratis.Bitcoin.Tests.Common.TestFramework;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Stratis.Bitcoin.IntegrationTests.Wallet
+{
+    public partial class Wallet_address_generation_and_funds_visibility : BddSpecification
+    {
+        private const string SendingWalletName = "senderwallet";
+        private const string ReceivingWalletName = "receivingwallet";
+        private const string WalletPassword = "password";
+        public const string AccountZero = "account 0";
+        private const string ReceivingNodeName = "receiving";
+        private const string SendingNodeName = "sending";
+
+        private SharedSteps sharedSteps;
+        private NodeGroupBuilder nodeGroupBuilder;
+        private IDictionary<string, CoreNode> nodeGroup;
+        private CoreNode sendingStratisBitcoinNode;
+        private CoreNode receivingStratisBitcoinNode;
+        private long walletBalance;
+
+        protected override void BeforeTest()
+        {
+            this.nodeGroupBuilder = new NodeGroupBuilder(Path.Combine(this.GetType().Name, this.CurrentTest.DisplayName));
+            this.sharedSteps = new SharedSteps();
+        }
+
+        protected override void AfterTest()
+        {
+        }
+
+        public Wallet_address_generation_and_funds_visibility(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        private void a_default_gap_limit_of_20()
+        {
+            this.nodeGroup = this.nodeGroupBuilder
+                .StratisPowNode(SendingNodeName).Start().NotInIBD()
+                .WithWallet(SendingWalletName, WalletPassword)
+                .StratisPowNode(ReceivingNodeName).Start().NotInIBD()
+                .WithWallet(ReceivingWalletName, WalletPassword)
+                .WithConnections()
+                .Connect(SendingNodeName, ReceivingNodeName)
+                .AndNoMoreConnections()
+                .Build();
+
+            this.sendingStratisBitcoinNode = this.nodeGroup[SendingNodeName];
+            this.receivingStratisBitcoinNode = this.nodeGroup[ReceivingNodeName];
+
+            var coinbaseMaturity = 1;
+
+            this.sendingStratisBitcoinNode.FullNode.Network.Consensus.CoinbaseMaturity = coinbaseMaturity;
+            this.receivingStratisBitcoinNode.FullNode.Network.Consensus.CoinbaseMaturity = coinbaseMaturity;
+
+            this.sharedSteps.MineBlocks(coinbaseMaturity + 1, this.sendingStratisBitcoinNode, AccountZero, SendingWalletName, WalletPassword);
+        }
+
+        private void a_wallet_with_funds_at_index_20_which_is_beyond_gap_limit()
+        {
+            ExtPubKey xPublicKey = this.GetExtendedPublicKey(ReceivingNodeName);
+            var recipientAddressBeyondGapLimit = xPublicKey.Derive(new KeyPath($"0/{20}")).PubKey.GetAddress(Network.RegTest);
```

Comment:
String interpolation is not needed in here.

Reply:
good point, I'll correct that